mlflow/langchain/api_request_parallel_processor.py

```diff
@@ -344,6 +345,11 @@ def _transform_request_json_for_chat_if_necessary(request_json, lc_model):
                 2. A boolean indicating whether or not the request was transformed from the OpenAI
                 chat format.
         """
+        # Avoid converting the request to LangChain's Message format if the chain
+        # is an AgentExecutor, as LangChainChatMessage might not be accepted by the chain
+        if isinstance(lc_model, AgentExecutor):
+            return request_json, False
+
```

Comment:
(Just filing a follow-up JIRA in fine) Shall we extract these input conversion things out of `APIRequest` to a separate utility module and add unit tests for them? This class seem to do too much more than it was supposed to do as a dataclass of API request object, resulting in low test coverage. I think it makes our life easier when we start to receive more change requests for RAG/agent handling post DAIS🤔

Reply:
Created JIRA! Will work on it later